lightning/src/ln/peer_handler.rs

```diff
@@ -953,34 +967,28 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> PeerManager<D
 				self.message_handler.chan_handler.handle_announcement_signatures(&peer.their_node_id.unwrap(), &msg);
 			},
 			wire::Message::ChannelAnnouncement(msg) => {
-				let should_forward = match self.message_handler.route_handler.handle_channel_announcement(&msg) {
+				if match self.message_handler.route_handler.handle_channel_announcement(&msg) {
 					Ok(v) => v,
 					Err(e) => { return Err(e.into()); },
-				};
-
-				if should_forward {
-					// TODO: forward msg along to all our other peers!
+				} {
+					should_forward = Some(wire::Message::ChannelAnnouncement(msg));
 				}
 			},
 			wire::Message::NodeAnnouncement(msg) => {
-				let should_forward = match self.message_handler.route_handler.handle_node_announcement(&msg) {
+				if match self.message_handler.route_handler.handle_node_announcement(&msg) {
 					Ok(v) => v,
 					Err(e) => { return Err(e.into()); },
```

Comment:
nit: a bit less verbose would be (here and elsewhere)

```suggestion
				if match self.message_handler.route_handler.handle_node_announcement(&msg)
				    .map_err(|e| e.into())?
```